### PR TITLE
Flash message fix for provider delete

### PR DESCRIPTION
**Pull request by jistefan**
Newer versions have ManageIQ instead CFME, this should cover the issue

{{pytest: cfme/tests/infrastructure/test_providers.py -v -k 'provider_crud'}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 4fb51e72c21837678adf2ca16603362382adf9c2:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/common/provider.py [hunk 1]
```diff
@@ -240,9 +240,13 @@ def delete(self, cancel=True):
             invokes_alert=True)
         sel.handle_alert(cancel=cancel)
         if not cancel:
-            flash.assert_message_match(
-                'Delete initiated for 1 {} Provider from the CFME Database'.format(
```
Comment (psav) [inline cfme/common/provider.py]: So, related to this, this is going to start cropping up in lots of places, appliance now has a cached ```product_name``` attribute. Feel free to use this. Most objects should have an appliance attribute now anyway as part of navmazing, so this should be come
```python
            flash.assert_message_match(
                'Delete initiated for 1 {} Provider from the {} Database'.format(
                    self.string_name, self.appliance.product_name))
```

**Comment by dajoRH** [comment]
Lint report for commit 8effcf809eced2ce2909cb18cae32a9f957d0b00:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 30855f1d2cbbe4acd637a89e036a96bbe2ae0971:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 120ec725c6a1fc2af8d12c183d4b22a22829a218:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**